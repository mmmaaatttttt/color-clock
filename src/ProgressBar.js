import React from "react";
import "./ProgressBar.css";

function ProgressBar({ gradientPercentage, color1, color2, color3 }) {
  const background = `linear-gradient(
    90deg, ${color1} 0%,${color2} ${gradientPercentage}%, ${color3} 100%
  )`;
  return (
    <div className="ProgressBar">
      <div className="ProgressBar--bar" style={{ background }}></div>
    </div>
  );
}

export default ProgressBar;
