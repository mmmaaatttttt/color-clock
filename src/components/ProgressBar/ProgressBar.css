.ProgressBar {
  width: 90%;
  padding: 0 2rem;
  background-color: white;
  border: 0.1rem solid #d1d1d1;
  border-radius: 0.2rem;
  height: 8rem;
  display: flex;
  flex-direction: column;
  justify-content: center;
  margin: 4rem;
  opacity: 1;
  transition: opacity 1s;
}

.ProgressBar.hidden {
  opacity: 0;
}

.ProgressBar--bar {
  height: 3rem;
  margin: 0 2rem;
  border-radius: .2rem;
  border: 0.1rem solid #333;
  position: relative;
}

.ProgressBar--caret {
  color: #333;
  position: absolute;
  bottom: -2rem;
  margin-left: -1rem;
  transition: left 1s linear;
}
