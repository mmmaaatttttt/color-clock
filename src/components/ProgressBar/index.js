import React from "react";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faCaretUp } from "@fortawesome/free-solid-svg-icons";
import "./ProgressBar.css";

function ProgressBar({
  gradientPercentage,
  caretPercentage,
  colors,
  hidden
}) {
  const [color1, color2, color3] = colors;
  const background = `linear-gradient(
    90deg, ${color1} 0%,${color2} ${gradientPercentage}%, ${color3} 100%
  )`;
  return (
    <div className={`ProgressBar ${hidden ? "hidden" : ""}`} data-testid="ProgressBar">
      <div className="ProgressBar--bar" style={{ background }}>
        <FontAwesomeIcon
          icon={faCaretUp}
          size="2x"
          className="ProgressBar--caret"
          style={{ left: `${caretPercentage}%` }}
        />
      </div>
    </div>
  );
}

export default ProgressBar;
