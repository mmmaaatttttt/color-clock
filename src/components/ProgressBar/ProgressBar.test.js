import React from "react";
import { render } from "@testing-library/react";
import ProgressBar from "./";

describe("basic ProgressBar tests", function () {
  it("renders without crashing", function () {
    render(<ProgressBar />);
  });

  it("matches snapshot", function () {
    const { asFragment } = render(<ProgressBar />);
    expect(asFragment()).toMatchSnapshot();
  });
});
