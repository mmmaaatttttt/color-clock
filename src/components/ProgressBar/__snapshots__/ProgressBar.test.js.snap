// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic ProgressBar tests matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="ProgressBar "
  >
    <div
      class="ProgressBar--bar"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-caret-up fa-w-10 fa-2x ProgressBar--caret"
        data-icon="caret-up"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 320 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M288.662 352H31.338c-17.818 0-26.741-21.543-14.142-34.142l128.662-128.662c7.81-7.81 20.474-7.81 28.284 0l128.662 128.662c12.6 12.599 3.676 34.142-14.142 34.142z"
          fill="currentColor"
        />
      </svg>
    </div>
  </div>
</DocumentFragment>
`;
